File: src/SmartSolar.Application/DTOs/Auth/LoginResponse.cs
/*
 * File: LoginResponse.cs
 * Project: Smart Solar Microgrid Trading System
 * Purpose: Shared project source file for the SE4040 EAD implementation.
 * Note: Keep this header and add/update method-level comments as the code evolves.
 */

using SmartSolar.Application.DTOs.Users;

namespace SmartSolar.Application.DTOs.Auth;

public sealed record LoginResponse(
    string AccessToken,
    DateTime ExpiresAtUtc,
    UserResponse User);

File: mobile/SmartSolarMobile/app/src/main/java/com/smartsolar/mobile/data/remote/dto/LoginResponse.java
package com.smartsolar.mobile.data.remote.dto;

public class LoginResponse {
    private String accessToken;
    private String expiresAtUtc;
    private UserResponse user;

    public String getAccessToken() { return accessToken; }
    public String getExpiresAtUtc() { return expiresAtUtc; }
    public UserResponse getUser() { return user; }
}
